caproto/threading/client.py

```diff
@@ -550,19 +552,25 @@ def command_loop(self):
                     cid = command.cid
                     try:
                         with self._search_lock:
-                            name, queue = unanswered_searches.pop(cid)
+                            name, queue, _ = unanswered_searches.pop(cid)
                     except KeyError:
                         # This is a redundant response, which the EPICS
                         # spec tells us to ignore. (The first responder
                         # to a given request wins.)
-                        if name in self.search_results:
```

Comment:
Yeppp. This would be a `NameError` on `name`, a problem I saw in logs yesterday. Gotta fix this (and much else) in the async clients too.

Reply:
Worse, if it has been through once it _works_ but tells you the wrong PV name!